File: Semestru 4/Ingineria sistemelor soft/Laboratoare/Lab 3/Teatru/src/main/java/com/example/baschetgui/cs/repositories/IRepository.java
package com.example.baschetgui.cs.repositories;

import com.example.baschetgui.cs.models.Entity;

import java.util.ArrayList;

public interface IRepository<E> {
    boolean save(E e);

    boolean delete(Entity e);

    E find(Entity e);

    boolean update(Entity e, E newE);

    ArrayList<E> getAll();
}

File: Semestru 4/Medii de proiectare si programare/Laboratoare/Basket/Basket/repositories/IRepository.cs
using Baschet.models;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Text;

namespace Baschet.repositories
{
    interface IRepository<E>
    {
        public bool save(E e);

        public bool delete(E e);

        public List<E> getAll();
    }
}
